example_concurrent_training_test.go

```diff
@@ -149,7 +149,7 @@ func prep() (x, y Value, bs []batch) {
 }
 
 func concurrentTraining(xV, yV Value, bs []batch, es int) {
-	threads := runtime.GOMAXPROCS(-1) // reserve one thread for the CPU locked thread
```

Comment:
The comment was misleading. `-1` doesn't change the current setting of max cpus.
https://golang.org/pkg/runtime/#GOMAXPROCS says, this call will go away, when scheduler improves, so I switched to NumCPU()

Reply:
Nice catch !